server/src/main/java/org/elasticsearch/index/mapper/BinaryRangeUtil.java

```diff
@@ -19,20 +19,51 @@
 
 package org.elasticsearch.index.mapper;
 
+import org.apache.lucene.document.InetAddressPoint;
+import org.apache.lucene.store.ByteArrayDataInput;
 import org.apache.lucene.store.ByteArrayDataOutput;
 import org.apache.lucene.util.BytesRef;
 import org.apache.lucene.util.NumericUtils;
+import org.elasticsearch.common.TriFunction;
 
 import java.io.IOException;
+import java.net.InetAddress;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Comparator;
 import java.util.List;
 import java.util.Set;
 
-enum BinaryRangeUtil {
+public enum BinaryRangeUtil {
```

Comment:
Can we extract LengthType in his own file and leave this class package protected ? The encoding should remain internal.

Reply:
So, I feel really bad about making `LengthType` a top level enum; it's very much an implementation detail of the range encoding.  The more I thought about it, the more I came to feel it really should be part of `RangeType`, and my only objection to putting it there in the first place was that `RangeFieldMapper` is already 1000 lines and defines half a dozen classes.  So I made `RangeType` a top level enum and put `LengthType` under that.  `RangeType` needs to be public anyway, so there's no increased API surface with this arrangement. 

This seems like the most natural refactoring to me, since `LengthType` is a direct function of `RangeType`, but I'm open to rolling that back and just making `LengthType` a top level if you feel strongly that's the right way to do this.  Thanks for the feedback!

Reply:
ok fine with me, thanks for explaining